### PR TITLE
fixes #375: Source Plugin for Kafka Connect

### LINKED ISSUE #375: Source Plugin for Kafka Connect
Explore options for building this - particularly for Aura.

Exposed TransactionEventHandlers not yet available- may have to be polling approach of some sort

### PR DESCRIPTION
**Pull request by conker84**
Fixes #375

This PR creates a Source module for the Kafka Connect Plugin

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - Implemented the feature
  - Added tests
  - Added docs

### CODE REVIEW

**Comment by dryhurst** [comment]
When can we expect this to be implemented for Aura users? Is there a timeline?

**Comment by conker84** [comment]
> Add docs:
> 
> To this page: https://neo4j.com/labs/kafka/4.0/configuration/
> 
> Deprecation warning up top, see deprecation warning shared earlier. Note which parts of this page are about the DB plugin, which I think is all of it.
> 
> Same comment on these pages:
> https://neo4j.com/labs/kafka/4.0/producer/
> https://neo4j.com/labs/kafka/4.0/docker/
> 
> Deprecation warning here: https://neo4j.com/labs/kafka/4.0/neo4j-cluster/#cluster_neo4j_plugin
> 
> In this section on remote clients, update bolt+routing:// => neo4j://
> https://neo4j.com/labs/kafka/4.0/neo4j-cluster/#cluster_remote_clients

@moxious makes sense add the same deprecation here: (https://neo4j.com/labs/kafka/4.0/consumer/) ?

File: doc/docs/modules/ROOT/pages/configuration.adoc [hunk 1]
```diff
@@ -1,4 +1,9 @@
 [#neo4j_configuration_system]
+
+[NOTE]
+The Neo4j Streams Plugin running inside the Neo4j database is deprecated and will not be supported after version 4.2 of Neo4j.
```
Comment (moxious) [inline doc/docs/modules/ROOT/pages/configuration.adoc:4]: 4.3 I think, not 4.2

File: doc/docs/modules/ROOT/pages/consumer.adoc [hunk 2]
```diff
@@ -10,6 +10,11 @@ Use this section to configure Neo4j to ingest the data from Kafka into Neo4j.
 --
 
 endif::env-docs[]
+
+[NOTE]
+The Neo4j Streams Plugin running inside the Neo4j database is deprecated and will not be supported after version 4.2 of Neo4j.
```
Comment (moxious) [inline doc/docs/modules/ROOT/pages/consumer.adoc:15]: 4.3

File: doc/docs/modules/ROOT/pages/docker.adoc [hunk 3]
```diff
@@ -1,5 +1,9 @@
 = Run with Docker
 
+[NOTE]
+The Neo4j Streams Plugin running inside the Neo4j database is deprecated and will not be supported after version 4.2 of Neo4j.
```
Comment (moxious) [inline doc/docs/modules/ROOT/pages/docker.adoc:4]: same thing again.  ;)

File: doc/docs/modules/ROOT/pages/producer.adoc [hunk 4]
```diff
@@ -1,5 +1,10 @@
 = Source: Neo4j -> Kafka
 
+[NOTE]
+The Neo4j Streams Plugin running inside the Neo4j database is deprecated and will not be supported after version 4.2 of Neo4j.
```
Comment (moxious) [inline doc/docs/modules/ROOT/pages/producer.adoc:4]: 4.3

**Review by moxious — CHANGES_REQUESTED** [review]
Mostly LGTM

But database plugin is deprecated for Neo4j >= 4.3, not 4.2.  In fact, this repo adds to the dockerfile the use of the stuff with 4.3.   It's really only going forward it's deprecated.

If you say 4.2, then existing users on 4.3 will get the wrong idea that they are already doing something wrong.  It's only deprecated forward.

File: doc/docs/modules/ROOT/pages/configuration.adoc (hunk 1 above)
Comment (conker84) [reply doc/docs/modules/ROOT/pages/configuration.adoc:4]: Sorry @moxious I just copy/paste the text in your comment here:
https://github.com/neo4j-contrib/neo4j-streams/pull/482#commitcomment-56258070

I'll replace this and all the all the notes below as well

Comment (conker84) [reply doc/docs/modules/ROOT/pages/configuration.adoc]: @moxious (I was wondering) given the comment above maybe makes sense to change the notes in this way:

```
[NOTE]
The Neo4j Streams Plugin running inside the Neo4j database is deprecated and will not be supported after version 4.3 (included) of Neo4j.
```

**Comment by conker84** [comment]
I'm closing this since with @moxious we decided to move the feature to version `4.1` (see #496)

**Pull request closed without merging**